presto-main/src/main/java/com/facebook/presto/execution/SqlTask.java

```diff
@@ -71,6 +71,7 @@
     private static final Logger log = Logger.get(SqlTask.class);
 
     private final TaskId taskId;
+    private final UUID taskInstanceUUID;
     private final String taskInstanceId;
```

Comment:
I went with maintaining both the string representation and the UUID so that the existing code which uses the string representation continues to do so and whatever extra cost of calling UUID.toString each time getTaskInstanceId is called is avoided. If removing the string representation is deemed cleaner I can take this out.